### ISSUE TITLE
Native test fails

### ISSUE THREAD

**Issue by Rafael-Ramblas** [styled-components/jest-styled-components#76]
Hello, noob jest tester here, so srry if my error is basic.
I'm trying to test a React Native Styled Component, but i got the error : 
`TypeError: component.type is not a function at Object.<anonymous> (__tests__/Foo.test.js:13:16)`
My file: Foo.test.js
```
import React from 'react'
import styled from 'styled-components/native'
import renderer from 'react-test-renderer'
import "jest-styled-components/native"

const Foo = styled.View`
    background-color: red;
`

test('it works', () => {
  const tree = renderer.create(<Foo />).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('background-color', 'red')
})
```

**Comment by Kureev** [comment]
Hi @Rafael-Ramblas!

There is no such thing as a "noob question" 😉 All questions are good!
Regarding the topic: I have a gut feeling that the underlying version of Jest has been updated which caused this regression. Need to have a deeper look. Will try to do my best to check it up tomorrow

**Comment by MicheleBertoli** [comment]
Thank you very much @Kureev.
I can confirm we updated all the deps since we added the matcher.
Feel free to ping me if you need any help/information.

**Comment by Kureev** [comment]
Interesting... While trying to reproduce this one, got another error:
```
λ ~/Projects/test →  yarn jest
yarn jest v0.24.4
$ "/Users/xamd/Projects/test/node_modules/.bin/jest"
 FAIL  ./index.test.js
  ● Test suite failed to run

    Cannot find module 'StyleSheet' from 'react-native-implementation.js'

      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:191:17)
      at Object.get StyleSheet [as StyleSheet] (node_modules/react-native/Libraries/react-native/react-native-implementation.js:95:29)
      at Object.<anonymous> (node_modules/styled-components/lib/native/index.js:42:67)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        1.992s
Ran all test suites.
error Command failed with exit code 1.
```

Will keep you posted

**Comment by Rafael-Ramblas** [comment]
The weird thing is, the snapshot is written correctly, but it throws this error

```
// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "red",
      },
      undefined,
    ]
  }
/>
`;
```

**Comment by Kureev** [comment]
I have a feeling that in one of the newest versions of Jest/React there was a breaking change that changed signature of custom matchers. As far as there are no release notes for Jest project, it is hard to find a root cause. 

As a side note: if you would like to help with investigation, I'll be very grateful if you share the code you have, so I can start from there.

**Comment by Kureev** [comment]
Ok, I have some news: seems I know what happens here:
- Fiber nodes has different structure than Stack ones
- Styled Components from version 2.0 started to use a new parser (is it correct, @MicheleBertoli?) which means that plugin should be adjusted accordingly

TL;DR: `toHaveStyleRule` should be rewritten

**Comment by MicheleBertoli** [comment]
Thank you very much for looking into this @Kureev.
I can confirm SC v2 changes the way style is generated.
There's a [utility](https://github.com/styled-components/jest-styled-components/blob/master/src/utils.js) to deal with the different versions of SC that might help.

**Comment by Kureev** [comment]
> I can confirm SC v2 changes the way style is generated.

Probably, it is a good thing. Seems we don't need to run parser directly, I can get an access to object with styles w/o it. Just need some time to compose a PR...

**Comment by Rafael-Ramblas** [comment]
Thanks for the help guys!!

**Comment by Kureev** [comment]
Does it work for you now, @Rafael-Ramblas?

**Comment by MicheleBertoli** [comment]
Hello @sreeram767, thanks for your comment.
Would you be able to provide more information?
When you say "the same issue" are you referring to the original error described in the first comment?
Which version of the package are you using?
